### PR TITLE
Audio falloff re-work, and increased audio range.

### PR DESCRIPTION
**Pull request by ghost**
:cl: Qustinnus
sound: Increases the range of all audio, with increased fall-off
code: Improves fall-off code and moves away from byond's inbuilt falloff as much as possible.
/:cl:

imo; the ss13 audio-scape is quite barren, you can only hear most things if you can see them, which in my opinion doesn't make much sense. This changes that so you can hear further away, but falloff is much higher, so in reality you will only hear things relatively quietly when they're out of sight.

This PR increases the hearing distance of most sound by 9, excluding sounds such as antag items that are meant to be used stealthily (If I missed any please mention it in the PR.).

This PR also replaces Byond's inbuilt falloff system with something I made, (And thanks to potato for helping me throw together a formula for it). This fall-off system makes sound fall off more naturally, with sounds being full volume within a certain range, and then softly falling off until they are completely quiet. This makes for a smoother transition between "This sound is full volume" and "I dont hear this sound". 

This PR will obviously have some balance concerns with things such as e-swords and other stealthy tools,as mentioned before I'll make sure all of them are still as short-range as possible.

samples:

https://streamable.com/qxoncw

https://streamable.com/artmte

### CODE REVIEW

**Comment by Qustinnus** [comment]
actually fuck it

**Comment by msgerbs** [comment]
The original audio falloff change was bad enough at letting validhunters beeline toward traitors.

**Comment by Qustinnus** [comment]
did you read the pr

**Comment by Qustinnus** [comment]
alright the bulk of it all is done. This definitely needs testmergign to see if the range is nice or not.

**Comment by Qustinnus** [comment]
made some samples

https://streamable.com/qxoncw

https://streamable.com/artmte

**Comment by msgerbs** [comment]
> 
> 
> did you read the pr

Yes, I see that you've reduced the range of some stealthy items, but it still means that if you're using anything not stealthy somebody will be able to hear from several screens away if I'm reading this right.

**Comment by LemonInTheDark** [comment]
If we're giving this a testmerge I want you there to tweak global values, and see what works well and what doesn't.

**Comment by Farquaar** [comment]
Honestly surprised that this wasn't in the game already. Good stuff!

**Comment by tralezab** [comment]
me likey

File: code/game/sound.dm [hunk 1]
```diff
@@ -38,13 +55,13 @@
...
-			M.playsound_local(turf_source, soundin, vol, vary, frequency, falloff, channel, pressure_affected, S)
+			M.playsound_local(turf_source, soundin, vol, vary, frequency, falloff_exponent, channel, pressure_affected, S, maxdistance, falloff_distance)
 	for(var/P in SSmobs.dead_players_by_zlevel[source_z])
 		var/mob/M = P
 		if(get_dist(M, turf_source) <= maxdistance)
-			M.playsound_local(turf_source, soundin, vol, vary, frequency, falloff, channel, pressure_affected, S)
+			M.playsound_local(turf_source, soundin, vol, vary, frequency, falloff_exponent, channel, pressure_affected, S, maxdistance, falloff_distance)
 
-/mob/proc/playsound_local(turf/turf_source, soundin, vol as num, vary, frequency, falloff, channel = 0, pressure_affected = TRUE, sound/S, distance_multiplier = 1)
+/mob/proc/playsound_local(turf/turf_source, soundin, vol as num, vary, frequency, falloff_exponent = SOUND_FALLOFF_EXPONENT, channel = 0, pressure_affected = TRUE, sound/S, max_distance, falloff_distance = SOUND_DEFAULT_FALLOFF_DISTANCE, distance_multiplier = 1)
```
Comment (LemonInTheDark) [inline code/game/sound.dm:64]: You'll need to document these variables and what they do, just gonna leave this [here](https://www.desmos.com/calculator/kedxfutvbi)

File: code/game/sound.dm [hunk 2]
```diff
@@ -69,7 +86,8 @@
 
 		distance *= distance_multiplier
 
-		S.volume -= max(distance - world.view, 0) * 2 //multiplicative falloff to add on top of natural audio falloff.
+		if(max_distance) //If theres no max_distance we're not a 3D sound, so no falloff.
+			S.volume -= (max(distance - falloff_distance, 0) ** falloff_exponent) / ((max_distance - falloff_distance) ** falloff_exponent) * S.volume
```
Comment (LemonInTheDark) [inline code/game/sound.dm:90]: max distance should be set to the max between it and distance, otherwise **very** dumb shit will happen

**Comment by Qustinnus** [comment]
@msgerbs I suppose thats true but what sounds would really give validhunters proper info? This will cloud up the sound space quite a bit so you wont be able to discern far away sounds that easy, and you'll also hear shit happening in a lot of directions so you wouldnt know which one is a traitor doing something.

**Comment by Qustinnus** [comment]
@LemonInTheDark 

testmerge went well, literaly noone complained, this can be reviewed/merged whenever. :)

**Comment by LemonInTheDark** [comment]
Aight, I'll get into it TomorrowTM

**Comment by Mickyan** [comment]
Does it work with instruments? I may have to undo their volume changes

**Comment by Qustinnus** [comment]
I think I removed the overrides for instruments, so yes this will work with them

**Comment by wesoda25** [comment]
Thank God now I don't have to hear the clowns music broadcasted through 2 walls

**Comment by Qustinnus** [comment]
you misunderstood : music instruments also have the extra range. I can nerf it though

**Review by Jack7D1 — APPROVED** [review]
I know this is feedback that no-one asked for; 
I believe it's worthwhile to state that the changed code is absolutely better than the old code. Regardless of what's being changed.
Additionally, while playing with the testmerge I notice that the soundscape feels much more alive, generally recommended for both code and gameplay quality..

**Comment by LemonInTheDark** [comment]
Today Tm

File: code/__DEFINES/sound.dm [hunk 3]
```diff
@@ -9,6 +9,17 @@
 #define CHANNEL_BUZZ 1017
 #define CHANNEL_BICYCLE 1016
 
+///Default range of a sound.
+#define SOUND_RANGE 17
+///default extra range for sounds considered to be quieter
+#define SHORT_RANGE_SOUND_EXTRARANGE -9
+///The range deducted from sound range for things that are considered silent / sneaky
+#define SILENCED_SOUND_EXTRARANGE -11
```
Comment (LemonInTheDark) [inline code/__DEFINES/sound.dm:17]: What's the difference between these 2 defines?

File: code/game/sound.dm [hunk 4]
```diff
@@ -1,4 +1,22 @@
...
+extrarange - modifier for sound range
+falloff_exponent - Rate of falloff for the audio
+frequency - playback speed of audio
+channel - The channel the sound is played at
+pressure_affected - Whether or not difference in pressure affects the sound (E.g. if you can hear in space)
+ignore_walls - Whether or not the sound can pass through walls.
+falloff_distance - Distance at which falloff begins.
+
+*/
+
```
Comment (LemonInTheDark) [inline code/game/sound.dm:18]: You should describe how changing some of these vars effects the sound, like the higher the exponent, the steeper the falloff, etc. That way it's easier to use without understanding the underlying system perfectly

**Review by LemonInTheDark — COMMENTED** [review]
Gave this a full read through, I didn't see anything horrendous, might want to describe how exactly some vars scale, but that's minor.

I love the concept of this pr, I'd caution you to not over correct on sound volume and presence, we want a healthy middle ground. From what I've seen this looks good tho, keep an eye on things the best you can, ~~play the game~~, all that.

**Comment by LemonInTheDark** [comment]
Yesterday TM

File: code/__DEFINES/sound.dm (hunk 3 above)
Comment (Qustinnus) [reply code/__DEFINES/sound.dm:17]: -2,

but all kidding aside, I used both where they sounded best, silenced is basically close to what we currently have for audio, and short range was just slightly further than that.

File: code/game/sound.dm (hunk 4 above)
Comment (Qustinnus) [reply code/game/sound.dm:18]: this better?

File: code/__DEFINES/sound.dm (hunk 3 above)
Comment (LemonInTheDark) [reply code/__DEFINES/sound.dm:17]: gottcha ~~I didn't see the minus on that 9~~

File: code/game/sound.dm (hunk 4 above)
Comment (LemonInTheDark) [reply code/game/sound.dm]: You should keep the above 1 thing

**Comment by LemonInTheDark** [comment]
Keep an eye on this, yadeyada

**Pull request merged**